test/python/algorithms/time_evolvers/classical_methods/test_scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,165 @@
+        It compares the solution with the first eigenstate of the hamiltonian.
+        """
+        print(type(expected_state))
+        expected_state_matrix = expected_state.to_matrix()
+
+        evolution_problem = TimeEvolutionProblem(hamiltonian, tau, initial_state)
+        classic_evolver = SciPyImaginaryEvolver(steps=300)
+        result = classic_evolver.evolve(evolution_problem)
+
+        with self.subTest("Amplitudes"):
+            np.testing.assert_allclose(
+                np.absolute(result.evolved_state.to_matrix()),
+                np.absolute(expected_state_matrix),
+                atol=1e-10,
+                rtol=0,
+            )
+
+        with self.subTest("Phases"):
+            np.testing.assert_allclose(
+                np.angle(result.evolved_state.to_matrix()),
+                np.angle(expected_state_matrix),
+                atol=1e-10,
+                rtol=0,
+            )
+
+    @data(
+        (
+            Zero ^ 5,
+            (X ^ (I ^ 4))
+            + (I ^ X ^ (I ^ 3))
+            + ((I ^ 2) ^ X ^ (I ^ 2))
+            + ((I ^ 3) ^ X ^ I)
+            + ((I ^ 4) ^ X),
+            5,
+        ),
+        (Zero, X, 1),
+    )
+    @unpack
+    def test_observables(self, initial_state: DictStateFn, hamiltonian: OperatorBase, nqubits: int):
+        """Tests if the observables are properly evaluated at each timestep."""
+
+        time_ev = 5.0
+        observables = {"Energy": hamiltonian, "Z": Z ^ nqubits}
+        evolution_problem = TimeEvolutionProblem(
+            hamiltonian, time_ev, initial_state, aux_operators=observables
+        )
+
+        classic_evolver = SciPyImaginaryEvolver(steps=300)
+        result = classic_evolver.evolve(evolution_problem)
+
+        z_mean, z_std = result.observables["Z"]
+
+        time_vector = result.observables["time"]
+        expected_z = 1 / (np.cosh(time_vector) ** 2 + np.sinh(time_vector) ** 2)
+        expected_z_std = np.zeros_like(expected_z)
+
+        np.testing.assert_allclose(z_mean, expected_z**nqubits, atol=1e-10, rtol=0)
+        np.testing.assert_allclose(z_std, expected_z_std, atol=1e-10, rtol=0)
+
+    def test_quantum_circuit_initial_state(self):
+        """Tests if the system can be evolved with a quantum circuit as an initial state."""
+        qc = QuantumCircuit(3)
+
+        qc.h(0)
+        qc.cx(0, range(1, 3))
+
+        evolution_problem = TimeEvolutionProblem(hamiltonian=X ^ X ^ X, time=1.0, initial_state=qc)
+        classic_evolver = SciPyImaginaryEvolver(steps=5)
+        result = classic_evolver.evolve(evolution_problem)
+        self.assertEqual(result.evolved_state, VectorStateFn(Statevector(qc)))
+
+    def test_error_time_dependency(self):
+        """Tests if an error is raised for time dependent hamiltonian."""
+        evolution_problem = TimeEvolutionProblem(
+            hamiltonian=X ^ X ^ X, time=1.0, initial_state=Zero, t_param=0
```

Comment:
Using `0` as a time parameter might be confusing.

Reply:
What would be a better candidate?

Reply:
How about `None`?

Reply:
But we are checking what happens when `t_param` is not None. If I input `None` we are not going to raise an exception.

Reply:
That looks like a bug in `TimeEvolutionProblem`, as the typehint (and default value) is `None` , in which case it shouldn't check if `time > 0` 😄